test/apps/data-filter/src/app.js

```diff
@@ -2,11 +2,17 @@
 /* eslint-disable no-console */
 import React, {Component} from 'react';
 import {render} from 'react-dom';
-import DeckGL, {COORDINATE_SYSTEM} from 'deck.gl';
+import DeckGL, {ScatterplotLayer, COORDINATE_SYSTEM} from 'deck.gl';
 
-import ScatterplotLayer from './scatterplot-layer';
+import DataFilterExtension from './data-filter-extension';
 import POINTS from './data-sample';
 
+import {extendLayer} from '@deck.gl/core';
+const FilteredScatterplotLayer = extendLayer(
```

Comment:
@heshan0131 The intention is that we will have a collection of extensions (e.g. data-filter, brushing) that can be used like this with any core layer.